static/js/compose.js

```diff
@@ -641,14 +640,17 @@ function validate_stream_message_post_policy(sub) {
 export function validation_error(error_type, stream_name) {
     let response;
 
-    const context = {};
-    context.stream_name = Handlebars.Utils.escapeExpression(stream_name);
-
     switch (error_type) {
         case "does-not-exist":
-            response = i18n.t(
-                "<p>The stream <b>__stream_name__</b> does not exist.</p><p>Manage your subscriptions <a href='#streams/all'>on your Streams page</a>.</p>",
-                context,
+            response = $t(
+                {
+                    defaultMessage:
+                        "<p>The stream <b>{stream_name}</b> does not exist.</p><p>Manage your subscriptions <z-link>on your Streams page</z-link>.</p>",
+                },
+                {
+                    stream_name: _.escape(stream_name),
+                    "z-link": (s) => `<a href='#streams/all'>${s}</a>`,
```

Comment:
How should one read `z-link`?  Is that short for `zulip-link`?  I see `<link>` in the FormatJS documentation.

Reply:
`z-link` is an arbitrary tag name we make up locally for this translation. I avoided `link` because readers or code prettifies could confuse it with the real `link` tag that means something different. I went with the `z-` scheme in the spirit of the [custom elements specification](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements-core-concepts), which requires at least one hyphen.

Reply:
OK.  Seems reasonable to me.